internal/mempool/mempool_test.go

```diff
@@ -2750,3 +2751,418 @@ func TestHandlesTAdds(t *testing.T) {
+			expectedAncestorFees: 1000,
+			expectedSizeBytes:    txALen,
+			expectedSigOps:       txANumSigOps,
+			ancestors:            []*dcrutil.Tx{txA},
+			descendants: map[chainhash.Hash]*dcrutil.Tx{
+				txD.MsgTx().TxHash(): txD,
+				txE.MsgTx().TxHash(): txE,
+			},
+			orderedAncestors: [][]*dcrutil.Tx{
+				{txA},
+			},
+		},
+		{
+			name:                 "Tx C",
+			subject:              txC,
+			expectedAncestorFees: 1000,
+			expectedSizeBytes:    txALen,
+			expectedSigOps:       txANumSigOps,
+			ancestors:            []*dcrutil.Tx{txA},
+			descendants: map[chainhash.Hash]*dcrutil.Tx{
+				txE.MsgTx().TxHash(): txE,
+			},
+			orderedAncestors: [][]*dcrutil.Tx{
+				{txA},
+			},
+		},
+		{
+			name:                 "Tx D",
+			subject:              txD,
+			expectedAncestorFees: 6000,
+			expectedSizeBytes:    txALen + txBLen,
+			expectedSigOps:       txANumSigOps + txBNumSigOps,
+			ancestors:            []*dcrutil.Tx{txA, txB},
+			descendants:          map[chainhash.Hash]*dcrutil.Tx{},
+			orderedAncestors: [][]*dcrutil.Tx{
+				{txA, txB},
+			},
+		},
+		{
+			name:                 "Tx E",
+			subject:              txE,
+			expectedAncestorFees: 11000,
+			expectedSizeBytes:    txALen + txBLen + txCLen,
+			expectedSigOps:       txANumSigOps + txBNumSigOps + txCNumSigOps,
+			ancestors:            []*dcrutil.Tx{txA, txB, txC},
+			descendants:          map[chainhash.Hash]*dcrutil.Tx{},
+			orderedAncestors: [][]*dcrutil.Tx{
+				{txA, txB, txC},
+				{txA, txC, txB},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		txHash := test.subject.Hash()
+		txDesc := harness.txPool.pool[*txHash]
+		miningView := harness.txPool.MiningView()
+
+		// Make sure transaction is not rejected between view instances.
+		if miningView.IsRejected(txHash) {
+			t.Fatalf("%v: expected test subject txn %v to not be rejected",
+				test.name, *txHash)
+		}
+
+		// Ensure stats are calculated before retrieving ancestors.
+		stat, hasStats := miningView.AncestorStats(txHash)
+		if !hasStats {
+			t.Fatalf("%v: expected mining view to have ancestor stats",
+				test.name)
+		}
+
+		if test.expectedAncestorFees != stat.Fees {
+			t.Fatalf("%v: expected subject txn to have bundle fees %v, got %v",
+				test.name, test.expectedAncestorFees, stat.Fees)
+		}
+
+		if test.expectedSizeBytes != stat.SizeBytes {
+			t.Fatalf("%v: expected subject txn to have SizeBytes=%v, got %v",
+				test.name, test.expectedSizeBytes, stat.SizeBytes)
+		}
+
+		if test.expectedSigOps != stat.NumSigOps {
+			t.Fatalf("%v: expected subject txn to have SigOps=%v, got %v",
+				test.name, test.expectedSigOps, stat.NumSigOps)
+		}
+
+		if len(test.ancestors) != stat.NumAncestors {
+			t.Fatalf("%v: expected subject txn to have NumAncestors=%v, got %v",
+				test.name, len(test.ancestors), stat.NumAncestors)
+		}
+
+		// Retrieving ancestors will cause the cached stats to be updated.
+		ancestors := miningView.Ancestors(txHash)
+		stat, _ = miningView.AncestorStats(txHash)
```

Comment:
Shouldn't there be some modification done before calling this and the stats be rechecked to ensure the cache is properly updated?  I can comment out the cache update and the tests don't fail which implies it's not being tested.

Reply:
Updated to assert the reference changed.